spring-context/src/test/java/org/springframework/context/annotation/configuration/AutowiredConfigurationTests.java

```diff
@@ -233,6 +234,15 @@ void testCustomPropertiesWithGenericContext() throws IOException {
 		context.close();
 	}
 
+	@Test
+	void testValueInjectionWithRecord() {
+		System.setProperty("recordBeanName", "recordBean");
+		GenericApplicationContext context = new AnnotationConfigApplicationContext(RecordBean.class);
+		context.refresh();
```

Comment:
FYI: a context cannot be refreshed multiple times.

This results in:

> java.lang.IllegalStateException: GenericApplicationContext does not support multiple refresh attempts: just call 'refresh' once

I will address this after merging the PR.